File: lib/models/car.dart
class Car {
  final int id;
  final String year;
  final int idModel;
  final String color;
  final String vin;
  final double price;
  final String? imagePath;

  Car({
    required this.id,
    required this.year,
    required this.idModel,
    required this.color,
    required this.vin,
    required this.price,
    this.imagePath,
  });

  factory Car.fromJson(Map<String, dynamic> json) {
    return Car(
      id: json['id_car'],
      year: json['year'],
      idModel: json['id_model'],
      color: json['color'],
      vin: json['vin'],
      price: double.tryParse(json['price'].toString()) ?? 0.0,
      imagePath: json['image_path']?.toString(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_car': id,
      'year': year,
      'id_model': idModel,
      'color': color,
      'vin': vin,
      'price': price,
      'image_path': imagePath,
    };
  }
}

File: car-sales-app/models/car.js


class Car {
    constructor(id, year, idModel, color, vin, price) {
        this.id = id;
        this.year = year;
        this.idModel = idModel;
        this.color = color;
        this.vin = vin;
        this.price = price;
    }

    static fromJson(json) {
        return new Car(
            json.id_car,
            json.year,
            json.id_model,
            json.color,
            json.vin,
            json.price
        );
    }

    toJson() {
        return {
            year: this.year,
            id_model: this.idModel,
            color: this.color,
            vin: this.vin,
            price: this.price
        };
    }
}

module.exports = Car;